src/Microsoft.Health.Fhir.Core/Features/Operations/Export/CreateExportRequestHandler.cs

```diff
@@ -0,0 +1,50 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using MediatR;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+using Microsoft.Health.Fhir.Core.Messages.Export;
+
+namespace Microsoft.Health.Fhir.Core.Features.Export
+{
+    public class CreateExportRequestHandler : IRequestHandler<CreateExportRequest, CreateExportResponse>
+    {
+        private IDataStore _dataStore;
+
+        public CreateExportRequestHandler(IDataStore dataStore)
+        {
+            EnsureArg.IsNotNull(dataStore, nameof(dataStore));
+
+            _dataStore = dataStore;
+        }
+
+        public async Task<CreateExportResponse> Handle(CreateExportRequest request, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            // TODO: Later we will add some logic here that will check whether a duplicate job already exists
+            // and handle it accordingly. For now we just assume all export jobs are unique and create a new one.
+
+            var jobRecord = new ExportJobRecord(request.RequestUri);
+
+            var responseCode = await _dataStore.UpsertExportJobAsync(jobRecord, cancellationToken);
```

Comment:
I think `UpsertExportJobAsync` should just throw exception if it fails to create the export job.